cirq/circuits/convert_to_cz_and_single_gates_test.py

```diff
@@ -146,6 +146,16 @@ def test_allow_partial_czs():
 
     assert circuit == c_orig
 
+    circuit2 = cirq.Circuit.from_ops(
+        cirq.TwoQubitMatrixGate((np.array([[1, 0, 0, 0],
+                                           [0, 1, 0, 0],
+                                           [0, 0, 1, 0],
+                                           [0, 0, 0, 1j]]))).on(q0, q1)
+    )
+    cirq.ConvertToCzAndSingleGates(allow_partial_czs=True
+                                   ).optimize_circuit(circuit2)
+
+    assert circuit2 == c_orig
```

Comment:
In this case there are two problems. First, the optimized circuit has a CZ**0.5 instead of the TwoQubitMatrixGate (as it should). Second, our 2-qubit decomposition is a bit, uh, lackadaisical when it comes to cleaning up single qubit operations that it generates and so there's a bunch of chaff operations.

Modify the check to assert that the list of 2-qubit operations in the circuit is of length 1 and that the operation in that list is a CZPowGate with an exponent near 0.5. `list(circuit.findall_operations(lambda e: isinstance(e, cirq.CZPowGate))) ...`.

Note: `cirq.testing.assert_same_circuit` gives much more informative error output.

Reply:
Thanks!